chapters/en/slides/module2/module2_02.Rmd

```diff
@@ -0,0 +1,279 @@
+gm = pd.read_csv(url, parse_dates=['year'])
+gm = gm[gm['country'].isin(gm[gm['year'] == '2018'].nlargest(10, ['population'])['country'])]
+gm
+```
+
+Notes:
+This dataset is made available via an URL
+and we can download and read it into pandas
+with `read_csv`.
+
+Here we also tell pandas to read in the `year` column as a date,
+rather than an integer,
+which is the default behavior.
+
+Remember that Altair uses the pandas data types
+to infer the data it is working with.
+Integers would show up on the Altair chart axes as `1,990, 2,000, etc`
+whereas dates are shown as `1990, 2000, etc`
+
+---
+
+## Using `info` to view dataframe columns
+
+```{python}
+gm.info()
+```
+
+Notes:
+When reading in a new dataset,
+it is always a good idea to have a look at a few rows
+like we did in the previous slide
+to get an idea of how the data looks.
+
+Another helpful practice is to use the `.info()` method
+to get an overview of the column types
+and see if there are any `NaNs` (missing values).
+
+---
+
+## A line chart is a good choice for trends over time
+
+```{python}
+import altair as alt
+
+
+alt.Chart(gm).mark_line().encode(
+    x='year',
+    y='population')
+```
+
+Notes:
+Since the data reaches all the way back to the 1800s,
+it would be really interesting to plot
+how the world population have been growing up until today.
+We could use a line plot for this
+as we learned in the first module.
+
+Why does the plot look so weird?
+Because for every year in the dataset,
+we have plotted the population for every single country
+and connected them with a vertical line.
+That's not what we want!
+
+---
+
+## The `sum` aggregate method will show us the trend of the entire world
+
+```{python}
+alt.Chart(gm).mark_line().encode(
+    x='year',
+    y='sum(population)')
+```
+
+Notes:
+We want to see how the **world** population has been changing.
+We can calculate this by summing up all the countries'
+populations for each year via the `sum` aggregation method in Altair.
+
+That looks much better!
+Interestingly,
+it seem like the world population have been growing in two distinct phases.
+We will talk more about that in a few slides,
+but first,
+let's recreate this chart as an area chart.
+
+---
+
+## Area charts are also effective at visualizing trends over time
+
+```{python}
+alt.Chart(gm).mark_area().encode(
+    x='year',
+    y='sum(population)')
+```
+
+Notes:
+Thanks to Altair's consistent visualization grammar,
+the modification we need to make to the code are minor.
+Replace `mark_line` with `mark_area` and we're done.
+
+For showing a singe trend over time,
+the choice between a line and area chart comes down to asthetics.
+They are both effective choices for communicating
+
+However,
+when visualizing the trends over time for multiple groups,
+lines and areas have different advantages,
+as we will see next.
+
+---
+
+## Area charts are preferred when the total of the groups is the most important
+
+```{python}
```

Comment:
Should the y-axis be something else here? Like cumulative sum? It's a bit different than how the data is interpreted on the next slide, but the axis labels are the same.